### ISSUE TITLE
Razer Blade Advanced 2019 Won't Connect to Wifi by Internal Wifi Card

### ISSUE THREAD

**Issue by RelentleSausage** [pop-os/pop#565]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="19.04"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.04"
VERSION_ID="19.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco

**Issue/Bug Description:**
Pop os will not connect to wifi through settings or cli. My wifi card seems to have support from linux in kernel version 5.1 as said in this article ( https://www.phoronix.com/scan.php?page=news_item&px=Intel-WiFi-6-AX200-Cyclone-Peak ). Although other users have had similar issues with this wifi card as said here ( https://unix.stackexchange.com/questions/518571/locating-drivers-for-intel-ax200-wireless-on-5-1-kernel ). After running *lspci -nns 02:00.0* the cli responds with *02:00.0 Network controller [0280]: Intel Corporation Device [8086:2723] (rev 1a)* . For for reference the support is added but is not in 5.1x yet. It might be supported with kernel 5.2 ( https://cateee.net/lkddb/web-lkddb/IWLWIFI.html ).

**Steps to reproduce (if you know):**
Use my Intel wifi card and attempt to connect to wifi

**Expected behavior:**
Wifi to work right out of the box and be able to be set in settings.

**Other Notes:**

Some Websites I found some info about my situation on:

https://www.phoronix.com/scan.php?page=news_item&px=Intel-WiFi-6-AX200-Cyclone-Peak

https://unix.stackexchange.com/questions/518571/locating-drivers-for-intel-ax200-wireless-on-5-1-kernel

https://cateee.net/lkddb/web-lkddb/IWLWIFI.html

Machine Specs:
Laptop name: Razer Blade 2019 Advanced
CPU: Intel Core i7-9750H
GPU: RTX 2080 Max Q
RAM: 16gb ddr4
SSD: Samsung 2tb 970 evo nvme ssd 
Wifi Card: Intel Wi-Fi 6 AX200 802.11ax

**Comment by R-Bower** [comment]
I'm also running into the same issue with my new Razer Blade 2019 Advanced.  Same wireless card.  Did you find a solution?

**Comment by RelentleSausage** [comment]
Unfortunately there isnt any solution, the wifi card is simply not supported due to there not being a driver. I found a work around by using a tp link usb wifi adapter (they work great on linux) but it’s not super convinient. I just ended up sending the blade back and opting for a system76 oryx pro with same specs and i like it a bit better.

**Comment by R-Bower** [comment]
Yeah, I wound up buying the TPLink USB.  I'm also considering returning the laptop.  How's the touchpad on your system76 oryx pro?  Razer's touchpad was the best I've ever used on a laptop and want something similar.

**Comment by RelentleSausage** [comment]
Actually it is very good. It’s a little bit sticky for the first hour or so when you get it but that goes away. Its quite large and I actually prefer it due to it having nice buttons and being about as large. It’s more reliable on linux I’ve found. The system76 is much more stable in the long run. I would highly reccomend it. The razer would stop going into the OS if I tried to change gpu’s as well and had some weird other unreliabilities on linux.

**Comment by e-m-m-a** [comment]
I too would like to use the AX200 card, but the kernel need to be 5.1. Any idea when this will be updated?

**Comment by RelentleSausage** [comment]
I dont have the blade anymore so probably wont be updated.

**Comment by R-Bower** [comment]
I returned my Razer and picked up an XPS 15 7590.  Supposedly it has a rebranded AX200 WiFi card, and it wasn't working out of the box on Linux Mint 19.1.  I upgraded the kernel to 5.2 and managed to get the WiFi working by following this guide: https://wiki.archlinux.org/index.php/Dell_XPS_15_7590
Follow the steps in the WiFi section.  I should note that I had to follow the optional step at the end: 

> Kernel versions 5.2.0 and 5.2.1 have issues with getting the WiFi module working . However, according to the same post, removing the file /lib/firmware/iwlwifi-cc-a0-48.ucode will help, which keeps the troublesome firmware verision 48 from being loaded and load the version 46 firmware instead.

**Comment by waaygone** [comment]
0

Today i will show you how to get your WiFI card working properly on you Razer Blade Pro 17 dual booted into any version of Linux OS Kali, Ubuntu etc.. And this will also work on any other razer blade model. My WiFI card is currently a Wi-Fi 6E AX210 Intel card.

    Download the WiFI drivers on the intel website for your particular WiFI Card in your computer https://www.intel.com/content/www/us/en/support/articles/000005511/wireless.html save it anywhere. I will save it in Downloads

    Open Terminal in Linux and go to directory where you downloaded the Intel WiFi driver and type: "Ls" to display everything in Downloads folder. Next type: tar zxvf iwlwifi-ty-59.601f3a66.0.tgz this will extract the folder.

    The folder should be extracted. Direct into the extracted folder using these commands: cd iwlwifi-cc-46.3cfab8da.0

    Next copy the .ucode file to the firmware folder listed in the lib folder sudo cp iwlwifi-cc-a0-46.ucode /lib/firmware

    NEXT STEP IS VERY IMPORTANT. You need to make sure your kernel is 5.12.0 or later which most people computers are not and that's why these steps above don't work for you! type: uname -a in the terminal and it should see 5.12.0 or later if not PLEASE follow step 6.

    If your kernel is not 5.12.0 or later you must upgrade your kernel via terminal. Follow and type all of these commands in the terminal step by step.

cd /tmp/

wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.12/amd64/linux-headers-5.12.0-051200_5.12.0-051200.202104252130_all.deb

wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.12/amd64/linux-headers-5.12.0-051200-generic_5.12.0-051200.202104252130_amd64.deb

wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.12/amd64/linux-image-unsigned-5.12.0-051200-generic_5.12.0-051200.202104252130_amd64.deb

wget -c https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.12/amd64/linux-modules-5.12.0-051200-generic_5.12.0-051200.202104252130_amd64.deb

sudo dpkg -i *.deb

    After these have been successfully completed you should be on the latest kernel necessary in order for your OS(Linux) to recognize you WiFi card(AX210 WiFi6E). Reboot and you should see the WiFi symbol. You are welcome!
